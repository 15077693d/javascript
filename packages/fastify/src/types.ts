import type { ClerkOptions } from '@clerk/backend';

export type ClerkFastifyOptions = Omit<ClerkOptions, 'apiKey'>;
