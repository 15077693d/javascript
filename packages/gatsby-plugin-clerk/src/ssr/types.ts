import { Session, User } from '@clerk/backend-core/src';
import type { GetServerDataProps } from "gatsby"
import { ServerGetToken } from '@clerk/types';

export type GetServerDataPropsWithAuth = GetServerDataProps & {
  auth?: {
    session: Session | undefined | null;
    userId: string | null;
    sessionId: string | null;
    getToken: ServerGetToken
  };
  user?: User | undefined | null;
}
