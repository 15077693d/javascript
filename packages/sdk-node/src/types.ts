import type { AuthenticateRequestOptions, AuthObject, Clerk, SignedInAuthObject } from '@clerk/backend';
import type { MultiDomainAndOrProxy } from '@clerk/types';
import type { NextFunction, Request, Response } from 'express';
import type { IncomingMessage } from 'http';

type LegacyAuthObject<T extends AuthObject> = Pick<T, 'sessionId' | 'userId' | 'actor' | 'getToken' | 'debug'> & {
  claims: AuthObject['sessionClaims'];
};

type BooleanOrURLFnToBoolean = boolean | ((url: URL) => boolean);

export type MiddlewareWithAuthProp = (
  // req: WithAuthProp<Request>
  req: Request,
  res: Response,
  next: NextFunction,
) => Promise<void>;

export type MiddlewareRequireAuthProp = (
  // req: RequireAuthProp<Request>,
  req: Request,
  res: Response,
  next: NextFunction,
) => Promise<void>;

export type LooseAuthProp = { auth: LegacyAuthObject<AuthObject> };
export type StrictAuthProp = { auth: LegacyAuthObject<SignedInAuthObject> };
export type WithAuthProp<T> = T & LooseAuthProp;
export type RequireAuthProp<T> = T & StrictAuthProp;

export type ClerkMiddleware = MiddlewareWithAuthProp | MiddlewareRequireAuthProp;
export type ClerkMiddlewareOptions = {
  onError?: (error: any) => unknown;
  authorizedParties?: string[];
  audience?: AuthenticateRequestOptions['audience'];
  jwtKey?: string;
  strict?: boolean;
  signInUrl?: string;
  /**
   * @experimental
   */
  experimental_isApiRoute?: BooleanOrURLFnToBoolean;
} & MultiDomainAndOrProxy;

export type ClerkClient = ReturnType<typeof Clerk>;

export type AuthenticateRequestParams = {
  clerkClient: ClerkClient;
  publishableKey?: string;
  secretKey?: string;
  /**
   * @deprecated Use `publishableKey` instead.
   */
  frontendApi?: string;
  /**
   * @deprecated Use `secretKey` instead.
   */
  apiKey?: string;
  req: IncomingMessage;
  options?: ClerkMiddlewareOptions;
};
