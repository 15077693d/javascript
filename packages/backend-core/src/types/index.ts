export * from './jwt';
export * from './core';
