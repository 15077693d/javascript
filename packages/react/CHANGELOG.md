# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

### [2.11.2-staging.0](https://github.com/clerkinc/javascript/compare/@clerk/clerk-react@2.11.1...@clerk/clerk-react@2.11.2-staging.0) (2022-02-15)

**Note:** Version bump only for package @clerk/clerk-react





### [2.11.1](https://github.com/clerkinc/javascript/compare/@clerk/clerk-react@2.11.1-staging.0...@clerk/clerk-react@2.11.1) (2022-02-14)

**Note:** Version bump only for package @clerk/clerk-react





### 2.11.1-staging.0 (2022-02-11)


### Features

* **clerk-sdk-node:** Deprecate Session named middleware, introduce withAuth, requireAuth ([4e69553](https://github.com/clerkinc/javascript/commit/4e695535e41fe7c135cbf303a0d021e7b7d30f7d))
