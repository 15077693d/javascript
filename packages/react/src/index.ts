export * from './contexts';
export * from './components';
export type { ClerkProp } from './types';
export { isMagicLinkError, MagicLinkErrorCode } from './errors';
export * from './hooks';
