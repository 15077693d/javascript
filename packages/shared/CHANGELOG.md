# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

### [0.0.16-alpha.4](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.16-alpha.3...@clerk/shared@0.0.16-alpha.4) (2022-03-23)

**Note:** Version bump only for package @clerk/shared

### [0.0.16-alpha.3](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.16-alpha.2...@clerk/shared@0.0.16-alpha.3) (2022-03-22)

**Note:** Version bump only for package @clerk/shared

### [0.0.16-alpha.2](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.16-staging.0...@clerk/shared@0.0.16-alpha.2) (2022-03-22)

**Note:** Version bump only for package @clerk/shared

### [0.0.16-alpha.1](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.16-staging.0...@clerk/shared@0.0.16-alpha.1) (2022-03-20)

**Note:** Version bump only for package @clerk/shared

### [0.0.16-alpha.0](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.16-staging.0...@clerk/shared@0.0.16-alpha.0) (2022-03-19)

**Note:** Version bump only for package @clerk/shared

### [0.0.15](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.15-staging.1...@clerk/shared@0.0.15) (2022-03-17)

**Note:** Version bump only for package @clerk/shared

### [0.0.15-staging.1](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.15-staging.0...@clerk/shared@0.0.15-staging.1) (2022-03-17)

**Note:** Version bump only for package @clerk/shared

### [0.0.13](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.12...@clerk/shared@0.0.13) (2022-03-11)

**Note:** Version bump only for package @clerk/shared

### [0.0.12](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.12-staging.0...@clerk/shared@0.0.12) (2022-03-09)

**Note:** Version bump only for package @clerk/shared

### [0.0.10](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.9...@clerk/shared@0.0.10) (2022-03-04)

**Note:** Version bump only for package @clerk/shared

### [0.0.9](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.8...@clerk/shared@0.0.9) (2022-03-04)

**Note:** Version bump only for package @clerk/shared

### [0.0.8](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.7...@clerk/shared@0.0.8) (2022-03-03)

**Note:** Version bump only for package @clerk/shared

### [0.0.7](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.6...@clerk/shared@0.0.7) (2022-03-02)

**Note:** Version bump only for package @clerk/shared

### [0.0.6](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.5...@clerk/shared@0.0.6) (2022-03-01)

### Bug Fixes

- **shared:** Phone input should start blank ([43f0b06](https://github.com/clerkinc/clerk_docker/commit/43f0b0603608866f6b9e0a37a284c0ea72c0004b))

### [0.0.5](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.5-staging.0...@clerk/shared@0.0.5) (2022-02-24)

**Note:** Version bump only for package @clerk/shared

### [0.0.5-staging.0](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.4-staging.0...@clerk/shared@0.0.5-staging.0) (2022-02-24)

**Note:** Version bump only for package @clerk/shared

### [0.0.4-staging.0](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.3-staging.0...@clerk/shared@0.0.4-staging.0) (2022-02-17)

**Note:** Version bump only for package @clerk/shared

### [0.0.3-staging.0](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.2...@clerk/shared@0.0.3-staging.0) (2022-02-15)

**Note:** Version bump only for package @clerk/shared

### [0.0.2](https://github.com/clerkinc/clerk_docker/compare/@clerk/shared@0.0.2-staging.0...@clerk/shared@0.0.2) (2022-02-14)

**Note:** Version bump only for package @clerk/shared

### 0.0.2-staging.0 (2022-02-11)

**Note:** Version bump only for package @clerk/shared
