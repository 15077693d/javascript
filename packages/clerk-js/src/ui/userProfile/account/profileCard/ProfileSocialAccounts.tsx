import { List } from '@clerk/shared/components/list';
import { ExternalAccountResource } from '@clerk/types';
import React from 'react';
import { svgUrl } from 'ui/common';
import { useCoreUser } from 'ui/contexts';
import { useNavigate } from 'ui/hooks';

export function ProfileSocialAccounts(): JSX.Element {
  const { navigate } = useNavigate();
  const user = useCoreUser();
  const verifiedAccounts = user.verifiedExternalAccounts;

  return (
    <List.Item
      className='cl-list-item'
      key='social-accounts'
      itemTitle='Social accounts'
      onClick={() => navigate('social-accounts')}
    >
      {verifiedAccounts.length === 0 ? (
        <div className='cl-empty-list-item'>None</div>
      ) : (
        <div className='cl-list-item-entry'>
          {verifiedAccounts.map((externalAccount: ExternalAccountResource) => (
            <div key={externalAccount.id}>
              <img
                alt={externalAccount.providerTitle()}
                src={svgUrl(externalAccount.provider)}
                className='cl-left-icon-wrapper'
              />

              {externalAccount.username || externalAccount.emailAddress}
            </div>
          ))}
        </div>
      )}
    </List.Item>
  );
}
