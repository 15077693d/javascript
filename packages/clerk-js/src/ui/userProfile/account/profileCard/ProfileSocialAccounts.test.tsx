import { renderJSON } from '@clerk/shared/testUtils';
import { ExternalAccountResource, UserResource } from '@clerk/types';
import * as React from 'react';

import { ProfileSocialAccounts } from './ProfileSocialAccounts';

const mockVerifiedExternalAccount = {
  id: '1',
  provider: 'google',
  emailAddress: 'clerk@clerk.dev',
  verification: { status: 'verified' },
  providerTitle: () => 'Google',
} as ExternalAccountResource;

const mockNavigate = jest.fn();
jest.mock('ui/hooks', () => ({
  useNavigate: jest.fn(() => {
    return {
      navigate: mockNavigate,
    };
  }),
}));

const mockIdentIsPrimary = jest.fn();
jest.mock('ui/contexts/CoreUserContext', () => {
  return {
    useCoreUser: (): Partial<UserResource> => {
      return {
        externalAccounts: [mockVerifiedExternalAccount],
        verifiedExternalAccounts: [mockVerifiedExternalAccount],
      };
    },
  };
});

describe('<ProfileSocialAccounts/>', () => {
  it('renders the ProfileSocialAccounts', () => {
    mockIdentIsPrimary.mockImplementationOnce(() => true);
    const tree = renderJSON(<ProfileSocialAccounts />);
    expect(tree).toMatchSnapshot();
  });
});
