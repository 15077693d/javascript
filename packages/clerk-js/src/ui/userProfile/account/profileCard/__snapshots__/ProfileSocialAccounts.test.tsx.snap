// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileSocialAccounts/> renders the ProfileSocialAccounts 1`] = `
<button
  className="cl-list-item elementContainer button hoverable"
  disabled={false}
  onClick={[Function]}
>
  <div
    className="title"
  >
    Social accounts
  </div>
  <div
    className="listItem line"
  >
    <div
      className="start"
    >
      <div
        className="cl-list-item-entry"
      >
        <div>
          <img
            alt="Google"
            className="cl-left-icon-wrapper"
            src="https://images.clerk.dev/static/google.svg"
          />
          clerk@clerk.dev
        </div>
      </div>
    </div>
    <div
      className="iconContainer"
    >
      <svg
        data-filename="../shared/assets/icons/arrow-right.svg"
      />
    </div>
  </div>
</button>
`;
